### PR TITLE
(34894): Wrong information in error message for overridden acc…

### LINKED ISSUE #34894: Wrong information in error message for overridden accessor
```ts
class Base {
    set data(value: number) {
        console.log("data changed to " + value);
    }
}

class Derived extends Base {
    data = 10;
}
```

This gives the error message:

```
Class 'Base' defines instance member accessor 'data', but extended class 'Derived' defines it as instance member property.
```

But there is clearly no

> instance member accessor 'data'

because `data` is an accessor on the prototype.

What about the following?

> 'data' is defined as an accessor in class 'Base', but is overridden here in 'Derived' as an instance property.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #34894 

cc @DanielRosenwasser

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -31814,9 +31814,10 @@ namespace ts {
                             continue;
                         }
                         if (basePropertyFlags !== SymbolFlags.Property && derivedPropertyFlags === SymbolFlags.Property) {
-                            errorMessage = Diagnostics.Class_0_defines_instance_member_accessor_1_but_extended_class_2_defines_it_as_instance_member_property;
+                            const errorMessage = Diagnostics._0_is_defined_as_an_accessor_in_class_1_but_is_overridden_here_in_2_as_an_instance_property;
+                            error(getNameOfDeclaration(derived.valueDeclaration) || derived.valueDeclaration, errorMessage, symbolToString(base), typeToString(baseType), typeToString(type));
                         }
-                        else if (basePropertyFlags === SymbolFlags.Property && derivedPropertyFlags !== SymbolFlags.Property) {
+                        if (basePropertyFlags === SymbolFlags.Property && derivedPropertyFlags !== SymbolFlags.Property) {
                             errorMessage = Diagnostics.Class_0_defines_instance_member_property_1_but_extended_class_2_defines_it_as_instance_member_accessor;
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: This one should probably be updated too

Comment (a-tarasyuk) [reply src/compiler/checker.ts]: @DanielRosenwasser updated

File: tests/baselines/reference/accessorsOverrideProperty.errors.txt [hunk 2]
```diff
@@ -18,7 +18,7 @@ tests/cases/conformance/classes/propertyMemberDeclarations/accessorsOverrideProp
          _secret = 11
         get p() { return this._secret } // error
             ~
-!!! error TS2611: Class 'C' defines instance member property 'p', but extended class 'D' defines it as instance member accessor.
+!!! error TS2611: 'p' is defined as an property in class 'C', but is overridden here in 'D' as an instance accessor.
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/accessorsOverrideProperty.errors.txt]: a property :)

Comment (DanielRosenwasser) [reply tests/baselines/reference/accessorsOverrideProperty.errors.txt]: `an instance accessor` -> `an accessor`

Comment (a-tarasyuk) [reply tests/baselines/reference/accessorsOverrideProperty.errors.txt]: :laughing:

**Pull request merged**